src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets

```diff
@@ -1,49 +1,71 @@
-                     ValidateWithPreviousVersion="$(ValidateWithPreviousVersion)"/>
+                     BaselineValidation="$(EnablePackageBaselineValidation)"
+                     BaselinePackageTargetPath="$(PackageValidationBaselinePath)" />
   </Target>
 
-  <Target Name="AddPackageDownload" BeforeTargets="CollectPackageDownloads" Condition="'$(ValidateWithPreviousVersion)' == 'true' and '$(PreviousPackagePath)' == ''">
-    <GetLastStablePackage Condition="'$(PreviousPackageVersion)' == ''"
-                          PackageId="$(PreviousPackageName)"
+  <!-- If package validation against a baseline is enabled and a local path to a baseline package
+       isn't specified, try to find the baseline package on the feeds. Achieving this by hooking
+       onto the CollectPackageDownloads extension point in cross-targeting builds. -->
+  <Target Name="AddPackageValidationBaselinePackageAsPackageDownload"
+          BeforeTargets="$(AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets)"
+          DependsOnTargets="_GetRestoreSettings"
+          Condition="'$(EnablePackageBaselineValidation)' == 'true' and
+                     '$(PackageValidationBaselinePath)' == '' and
+                     '$(IsCrossTargetingBuild)' == 'true'">
+    <!-- TODO: Why not use PackageId instead of AssemblyName? -->
+    <PropertyGroup>
+      <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(AssemblyName)'))</PackageValidationBaselineName>
+    </PropertyGroup>
+
+    <GetLastStablePackage PackageId="$(PackageValidationBaselineName)"
                           PackageVersion="$(Version)"
-                          NugetFeeds="$(_OutputSources)">
-      <Output TaskParameter="LastStableVersion" PropertyName="PreviousPackageVersion" />
+                          NugetFeeds="$(_OutputSources)"
+                          Condition="'$(PackageValidationBaselineVersion)' == ''">
+      <Output TaskParameter="LastStableVersion" PropertyName="PackageValidationBaselineVersion" />
     </GetLastStablePackage>
 
-    <ItemGroup Condition="'$(PreviousPackageVersion)' != ''">
-      <PackageDownload Include="$(PreviousPackageName)" Version="[$(PreviousPackageVersion)]" />
+    <ItemGroup>
+      <PackageDownload Include="$(PackageValidationBaselineName)"
+                       Version="[$(PackageValidationBaselineVersion)]" />
     </ItemGroup>
 
-    <WriteLinesToFile File="$(PreviousPackageVersionFilePath)"
-                      Lines="$(PreviousPackageVersion)"
+    <!-- Write the baseline path to disk for it to be accessible during Pack which
+         usually doesn't share evaluation (nodes) with restore.
+         PackageDownload doesn't support GeneratePathProperty which is why the path needs
+         to be written into a separate file than the NuGet generated props file:
+         https://github.com/NuGet/Home/issues/8476. -->
+    <WriteLinesToFile File="$(PackageValidationIntermediateBaselineFile)"
```

Comment:
You could potentially avoid the read side of this by emitting a props file that matches the same globbing convention msbuild uses for loading nuget's generated props/targets.  https://github.com/dotnet/msbuild/blob/5b9216a75e98e19eba84e04a5f30bd35a68f317a/src/Tasks/Microsoft.Common.props#L63

Reply:
I would prefer that as well, i.e.: `system.drawing.common.csproj.packagevalidation.g.props`

```xml
<Project>
  <PropertyGroup>
    <PackageValidationBaselineName Condition="'$(PackageValidationBaselineName)' == ''">foo</PackageValidationBaselineName>
    <PackageValidationBaselineVersion Condition="'$(PackageValidationBaselineVersion)' == ''">1.0.0</PackageValidationBaselineVersion>
  </PropertyGroup>
</Project>
```xml